### ISSUE TITLE
Tutorials for more advanced topics with R

### ISSUE THREAD

**Issue by fmichonneau** [ropensci/unconf16#35]
Last year's discussion about using R packages to facilitate reproducibility was fruitful, and led to the creation a https://github.com/ropensci/rrrpkg 

Along the same lines of using R for reproducible research, I would love to have discussions leading to tutorials/lessons about:
- how to get started with Docker
- how to setup continuous integration and coverage testing (related to #6)
- using remake 

also related to #23

**Comment by cboettig** [comment]
:+1:  

One of the themes in last year's discussion around https://github.com/ropensci/rrrpkg involved tricks & workarounds in adapting the R package format for  academic / research projects.  One piece we didn't really hit on then but is a bit more mature now is the use of [rticles](https://github.com/rstudio/rticles) as part of that workflow.  Might be some discussion/exploration on how much that helps facilitate things or what could to be done still to reduce the friction involved.

**Comment by jennybc** [comment]
Building on some of @fmichonneau's themes:

I would love to hear how other non-Windows users solve their package's or analytical pipeline's Windows problems. I don't think, e.g., appveyor, is a genuine solution. Too arm's length, too canned. Do most people just break down and set up a VM? I want details!

**Comment by benmarwick** [comment]
Yes, I'm very interested to see how people are solving problems of file organisation and reproducibility in their real-world work. In last year's discussion there was disagreement about the value of the package structure for research projects, so it would be great to learn more about what people are doing when not using a package structure. I'd love to see some lightning talks where people give a quick overview of how they organised their files and workflow to make their most recent project reproducible.

**Comment by HeidiSeibold** [comment]
I'd be very interested in that discussion :+1:

**Comment by maelle** [comment]
"how to setup continuous integration and coverage testing" great idea! A tutorial with lots of pictures would be awesome! I think I still remember issues I had:
- I had no idea what CI was to begin with.
- How to customize the config files, for using packages from source, for using different versions, etc.
- How to use a secure key from Github for package installation (instead of copying a non secure key for everyone to see, hum)
- How to use lintr in Travis and make the lintr bot send you notifications (nor not send you notifications).
- Travis vs. Appveyor.
- How exactly to get the badges.

Another point that comes to mind is: what are artifacts on Appveyor and why use them? I see what they are but I don't know why I would ever use them, but I've read it's a strength of Appveyor.

**Comment by leeper** [comment]
Sponsor artifacts could be used to store a package binary or built PDF of
documentation. I suspect no one actually uses that feature, though, even
though it could be useful.

**Comment by ttimbers** [comment]
+1 to a discussion on the topics @fmichonneau suggested. In particular, I could contribute to a discussion and contribute to a lesson on getting started with R and Docker. 

@jennybc my current solution to solve package and software management between different machines (Windows, Linux, OS X) is to use Docker (which admittedly runs in a VM on OS X and Windows). I built an ubuntu Docker image with R installed for this purpose. I have documented this in a blog post [here](http://tiffanytimbers.com/Using-docker-for-reproducible-research/). I am aware of [Rocker](https://hub.docker.com/r/rocker/r-base/), but at the time I looked into it their images didn't seem to allow me to run R scripts from the command line (in practice this is how I run the majority of my code). Does anyone else have experience with R & Docker, and/or Rocker/r-base? If so, I would be interested to hear how you are using it.
